### PR TITLE
Drop columns in groupby.size

### PR DESCRIPTION
**Pull request by phofl**
We don't need any columns for size, so drop all off them

### CODE REVIEW

File: dask_expr/_groupby.py [hunk 1]
```diff
@@ -533,6 +533,24 @@ class Size(SingleAggregation):
     groupby_chunk = M.size
     groupby_aggregate = M.sum
 
+    def _simplify_down(self):
+        if (
+            self._slice is not None
+            and not isinstance(self._slice, list)
+            or self.frame.ndim == 1
+        ):
+            # Single projections influence the result and are allowed
```
[the remark this answers is not in the document]
Comment (phofl) [reply dask_expr/_groupby.py]: df.groupby().y.size() will name the resulting series y, df.groupby().size() will have None as name

I clarified the comment

[the remark this answers is not in the document]
Comment (hendrikmakait) [reply dask_expr/_groupby.py]: Thanks for clarifying the comment!

File: dask_expr/_groupby.py [hunk 2]
```diff
@@ -533,6 +533,26 @@ class Size(SingleAggregation):
     groupby_chunk = M.size
     groupby_aggregate = M.sum
 
+    def _simplify_down(self):
+        if (
+            self._slice is not None
+            and not isinstance(self._slice, list)
+            or self.frame.ndim == 1
+        ):
+            # Scalar slices influence the result and are allowed, e.g. the name of
```
Comment (hendrikmakait) [inline dask_expr/_groupby.py:542]: IIUC, the name is the only difference
```suggestion
            # Scalar slices influence the result and are allowed, i.e., the name of
```

**Review by hendrikmakait — APPROVED** [review]
Thanks, @phofl!

**Pull request merged**